src/internal/connector/discovery/discovery.go

```diff
@@ -0,0 +1,83 @@
+package discovery
+
+import (
+	"context"
+
+	msgraphgocore "github.com/microsoftgraph/msgraph-sdk-go-core"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	msuser "github.com/microsoftgraph/msgraph-sdk-go/users"
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/support"
+)
+
+const (
+	userSelectID            = "id"
+	userSelectPrincipalName = "userPrincipalName"
+	userSelectDisplayName   = "displayName"
+)
+
+func Users(ctx context.Context, gs graph.Service, tenantID string) ([]models.Userable, error) {
+	users := make([]models.Userable, 0)
+
+	options := &msuser.UsersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &msuser.UsersRequestBuilderGetQueryParameters{
+			Select: []string{userSelectID, userSelectPrincipalName, userSelectDisplayName},
+		},
+	}
+
+	response, err := gs.Client().Users().Get(ctx, options)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"retrieving resources for tenant %s: %s",
+			tenantID,
+			support.ConnectorStackErrorTrace(err),
+		)
+	}
+
+	iter, err := msgraphgocore.NewPageIterator(response, gs.Adapter(),
+		models.CreateUserCollectionResponseFromDiscriminatorValue)
+	if err != nil {
+		return nil, errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+	}
+
+	var iterErrs error
+
+	callbackFunc := func(item interface{}) bool {
+		u, err := parseUser(item)
+		if err != nil {
+			iterErrs = support.WrapAndAppend("discovering users: ", err, iterErrs)
+			return true
+		}
+
+		users = append(users, u)
+
+		return true
+	}
+
+	if err := iter.Iterate(ctx, callbackFunc); err != nil {
+		return nil, errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+	}
```

Comment:
@dadams39 - the linked call is a for a Delta API (which PageIterator) doesn't support I believe. What is the change you're requesting and how does that help?

Reply:
Line 40 suggests that we are using an iterator.

Reply:
@dadams39 and I synced on this offline. We have a slight preference in the exchange/onedrive code for using the other style but that is related to either using `Delta` APIs or `DeltaTokens` which the page iterator doesn't support.

Since we don't have a use case for those in the user query - going to stick with this.